drivers/stepper/gpio_stepper/common/include/gpio_stepper_common.h

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright 2024 Fabian Blatz <fabianblatz@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVER_STEPPER_GPIO_STEPPER_COMMON_H_
+#define ZEPHYR_DRIVER_STEPPER_GPIO_STEPPER_COMMON_H_
+
+/**
+ * @brief Stepper Driver APIs
+ * @defgroup gpio_stepper Stepper Driver APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/stepper.h>
+#include <zephyr/drivers/counter.h>
+
+#include "stepper_timing_source.h"
+
+/**
+ * @brief Common GPIO stepper config.
```

Comment:
```suggestion
 * @brief Common GPIO motion controller config.
```
Think about renaming everything related to this to GPIO motion controller, which is admittedly quite a lot of stuff.
Everything here is about motion controllers, so I feel the names should reflect this.